src/incentives/AffinePassBridge.sol

```diff
@@ -0,0 +1,185 @@
+
+    // Event emitted when a message is sent to another chain.
+    event BridgeRequest(
+        bytes32 indexed messageId,
+        uint64 indexed destinationChainSelector,
+        address sender,
+        uint256 id 
+    );
+
+    // Event emitted when a message is received from another chain.
+    event BridgeReciept(
+        bytes32 indexed messageId,
+        uint64 indexed sourceChainSelector,
+        address receiver,
+        uint256 id
+    );
+
+    modifier onlyWhitelistedDestinationChain(uint64 _destinationChainSelector) {
+        if (!whitelistedDestinationChains[_destinationChainSelector])
+            revert DestinationChainNotWhitelisted(_destinationChainSelector);
+        _;
+    }
+
+    modifier onlyWhitelistedSourceChain(uint64 _sourceChainSelector) {
+        if (!whitelistedSourceChains[_sourceChainSelector])
+            revert SourceChainNotWhitelisted(_sourceChainSelector);
+        _;
+    }
+
+    modifier onlyWhitelistedSenders(address _sender) {
+        if (!whitelistedSenders[_sender]) revert SenderNotWhitelisted(_sender);
+        _;
+    }
+
+    constructor(address _affinePass, 
+                address router, 
+                uint64[] memory destinationChainSelectors, 
+                uint64[] memory sourceChainSelectors
+                ) CCIPReceiver(router) {
+        affinePass = AffinePass(_affinePass);
+        for (uint i = 0; i < destinationChainSelectors.length; i++) {
+            whitelistedDestinationChains[destinationChainSelectors[i]] = true;
+        }
+        for (uint i = 0; i < sourceChainSelectors.length; i++) {
+            whitelistedSourceChains[sourceChainSelectors[i]] = true;
+        }
+    }
+
+    fallback() external payable { }
+    receive() external payable { }
+
+    function setAffinePassAddress(address _affinePass) external onlyOwner {
+        affinePass = AffinePass(_affinePass);
+    }
+
+    function setPaused(bool _paused) external onlyOwner {
+        paused = _paused;
+    }
+
+    function setChainReciever(uint64 _chainSelector, address _reciever) external onlyOwner {
+        chainReciever[_chainSelector] = _reciever;
+    }
+
+    function ccipFee(
+        uint64 destinationChainSelector
+    ) external view returns (uint256) {
+        Client.EVM2AnyMessage memory message = _buildCCIPMessage(
+            chainReciever[destinationChainSelector],
+            address(0),
+            1,
+            address(0)
+        );
+
+        return IRouterClient(i_router).getFee(
+            destinationChainSelector,
+            message
+        );
+    }
+
+    function bridgePass(
+        uint64 destinationChainSelector,
+        address receiver,
+        uint256 id
+    ) external payable onlyWhitelistedDestinationChain(destinationChainSelector) returns (bytes32 messageId) {
+        require(affinePass.ownerOf(id) == msg.sender, "Not owner of token");
+        require(!paused, "Bridging is paused");
```

Comment:
We should use custom errors, instead of mixing custom errors with `require` statements.